lib/src/git.rs

```diff
@@ -364,15 +364,19 @@ pub fn fetch(
     fetch_options.proxy_options(proxy_options);
     let callbacks = callbacks.into_git();
     fetch_options.remote_callbacks(callbacks);
-    if globs.iter().any(|g| g.contains(|c| ":^".contains(c))) {
-        return Err(GitFetchError::InvalidGlob);
-    }
-    // At this point, we are only updating Git's remote tracking branches, not the
-    // local branches.
-    let refspecs = globs
-        .iter()
-        .map(|glob| format!("+refs/heads/{glob}:refs/remotes/{remote_name}/{glob}"))
-        .collect_vec();
+    let refspecs = if let Some(globs) = globs {
+        if globs.iter().any(|g| g.contains(|c| ":^".contains(c))) {
+            return Err(GitFetchError::InvalidGlob);
+        }
+        // At this point, we are only updating Git's remote tracking branches, not the
+        // local branches.
+        globs
+            .iter()
+            .map(|glob| format!("+refs/heads/{glob}:refs/remotes/{remote_name}/{glob}"))
+            .collect_vec()
+    } else {
+        vec![]
```

Comment:
Could you add a comment here saying that git2 interprets an empty list as saying that we should use the default.

Their documentation says that it means to "use the base refspecs". I'm not sure what that actually means. If it means that it respects the user's config (or system config), then we should probably always provide an explicit refspec here. But that's out of scope for this PR.

Reply:
I have added as a fifth commit (`git: use our own default refspec`): `*` is the default glob if none are presents. It will ensure we import `refs/heads/*` as `refs/remotes/REMOTENAME/*`.